core/src/wsv.rs

```diff
@@ -487,7 +442,7 @@ impl WorldStateView {
 
         for event in data_events {
             self.world.triggers.handle_data_event(event.clone());
-            self.produce_event(event);
+            self.events_buffer.borrow_mut().push(event.into());
```

Comment:
Maybe we could do this with single `borrow_mut().extend()`?

Reply:
Maybe? But doesn't that make the `.into()` harder?

Reply:
.map(Into::into) should do the job